### DISCUSSION TITLE
neo-tree and ftdetect

### DISCUSSION THREAD

**Discussion by gonzus** [LazyVim/LazyVim#1250] [Q&A] [answered]
I have a file `after/ftdetect/do.lua` with the following contents:
```lua
vim.api.nvim_create_autocmd(
  {
    "BufNewFile",
    "BufRead",
  },
  {
    pattern = "*.do",
    callback = function()
      local buf = vim.api.nvim_get_current_buf()
      vim.api.nvim_buf_set_option(buf, "filetype", "sh")
    end
  }
)
```
When I open any `*.do` file, its `filetype` is now correctly set to `sh`. But if I open `neo-tree`, all `*.do` files are shown with a `*` icon, whereas `*.sh` files are shown with the correct icon for a shell script:
<img width="246" alt="Screenshot 2023-07-29 at 11 48 18" src="https://github.com/LazyVim/LazyVim/assets/208546/aab42dd4-d908-44ad-984b-6f98bd7a1002">

Moving the file from `after/ftdetect/do.lua` to `ftdetect/do.lua` doesn't seem to have any effect on the icon.

How can the `neo-tree` icon be changed to be consistent with the `filetype`?

**Comment by dpetka2001** [comment, marked as answer]
```lua
  -- Modify `nvim-web-devicons`
  {
    "nvim-tree/nvim-web-devicons",
    opts = {
      override_by_extension = {
        ["do"] = {
          icon = "",
          color = "#89e051",
          cterm_color = "113",
          name = "Do",
        },
      },
    },
  },
```
You can take a look at `~/.local/share/nvim/lazy/nvim-web-devicons/lua/nvim-web-devicons.lua` to see the icons and colors different filetypes are defined and use whatever you like. I just used the same as `bash` in my example given.

**Reply by gonzus** [reply]
Ok, I see. I thought there would be a way to have a centralized definition for what a file type is, which would make everything behave consistently, including icons. But this will do nicely. Thanks!

**Reply by dpetka2001** [reply]
It's because `neo-tree` uses `web-devicons` to show the icons that are provided for filetypes etc(neo-tree only provides some icons for the internal sources it uses from the little that I've looked around). So the setup of `web-dev-icons` must be customized to provide you with the icons you want for different filetypes. The centralized definition for what a filetype is, is what you've already tried and it works for Neovim itself. Unfortunately, I don't think Neovim sets icons for the filetypes itself. It relies on other plugins, such as `web-devicons`(I might be wrong, this is just a guesstimate from my experience). So you must configure those plugins that provide them.

**Reply by gonzus** [reply]
Understood. I guess this might be a nice idea for a plugin that unifies these settings. Thanks again!
